packages/cli/src/templates/expo.ts

```diff
@@ -0,0 +1,277 @@
+import chalk from "chalk";
+import { Listr } from "listr2";
+import { existsSync } from "node:fs";
+import { writeFile } from "node:fs/promises";
+import { TemplateOptions } from "../commands/create";
+import { Template } from "./template";
+
+import { exec } from "node:child_process";
+import { promisify } from "node:util";
+const execAsync = promisify(exec);
+export interface ExpoContext {
+  options: TemplateOptions;
+}
+
+export const Expo: Template = {
+  name: "expo",
+  description: "A new expo typescript project with gluestack-ui renderer",
+  async handler(options) {
+    await new Listr<ExpoContext>([
+      {
+        title: "Create directory",
+        task: async ({ options: { cwd } }, task) => {
+          task.title += ` ${cwd}`;
+          if (existsSync(cwd)) {
+            throw new Error(`Directory ${cwd} already exists`);
+          }
+          await execAsync(
+            `npx create-expo-app ${cwd} -t expo-template-blank-typescript`,
+          );
+        },
+      },
+      {
+        title: "Create Expo project",
+        task: async ({ options: { cwd, name } }) => {
+          await writeFile(`${cwd}/app.json`, APP_JSON_CONTENT(name), "utf8");
```

Comment:
This file is created by `create-expo-app`.
It should not be overriden blindly IMHO, as it wiil be very fragile in light of `create-expo-app` tweaks.
We should modify it to add the modifications required.
In this case I believe the only thing changed is `"jsEngine": "jsc",`